xarray/core/dataarray.py

```diff
@@ -2767,25 +2774,56 @@ def stack(
         Examples
         --------
         >>> arr = xr.DataArray(
-        ...     np.arange(6).reshape(2, 3),
-        ...     coords=[("x", ["a", "b"]), ("y", [0, 1, 2])],
+        ...     np.arange(12).reshape(2, 3, 2),
+        ...     coords=[("x", ["a", "b"]), ("y", [0, 1, 2]), ("z", ["alpha", "beta"])],
         ... )
         >>> arr
-        <xarray.DataArray (x: 2, y: 3)>
-        array([[0, 1, 2],
-               [3, 4, 5]])
+        <xarray.DataArray (x: 2, y: 3, z: 2)>
+        array([[[ 0,  1],
+                [ 2,  3],
+                [ 4,  5]],
+
+               [[ 6,  7],
+                [ 8,  9],
+                [10, 11]]])
         Coordinates:
           * x        (x) <U1 'a' 'b'
           * y        (y) int64 0 1 2
-        >>> stacked = arr.stack(z=("x", "y"))
-        >>> stacked.indexes["z"]
+          * z        (z) <U5 'alpha' 'beta'
+        >>> stacked = arr.stack(newdim=("x", "y"))
+        >>> stacked
+        <xarray.DataArray (z: 2, newdim: 6)>
+        array([[ 0,  2,  4,  6,  8, 10],
+               [ 1,  3,  5,  7,  9, 11]])
+        Coordinates:
+          * z        (z) <U5 'alpha' 'beta'
+          * newdim   (newdim) object MultiIndex
+          * x        (newdim) <U1 'a' 'a' 'a' 'b' 'b' 'b'
+          * y        (newdim) int64 0 1 2 0 1 2
+        >>> stacked.indexes["newdim"]
         MultiIndex([('a', 0),
                     ('a', 1),
                     ('a', 2),
                     ('b', 0),
                     ('b', 1),
                     ('b', 2)],
-                   name='z')
+                   name='newdim')
+        >>> inverted_stack = arr.stack({"newdim": "z"}, invert=True)
```

Comment:
I think a clearer example would be exactly the same as https://github.com/pydata/xarray/pull/8332/files#diff-386f35dda14ca12d315f2eb71f0ab35d7fdc73419b5dc5175ea5720f3206903bR2793, but with the addition of `invert=True` (or whatever construction we end up using).

(If copying the new output is annoying, we can use pytest-accept...)